src/commands/channels/create.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2022, Nuance, Inc. and its contributors.
+ * All rights reserved.
+ *
+ * This source code is licensed under the Apache-2.0 license found in
+ * the LICENSE file in the root directory of this source tree.
+ */
+
+import {flags} from '@oclif/command'
+import makeDebug from 'debug'
+
+import * as MixFlags from '../../utils/flags'
+import * as ChannelsAPI from '../../mix/api/channels'
+import MixCommand from '../../utils/base/mix-command'
+import {DomainOption} from '../../utils/validations'
+import {ChannelModalities, ChannelModality, ChannelsCreateParams} from '../../mix/api/channels-types'
+import {asArray} from '../../utils/as-array'
+import {MixClient, MixResponse, MixResult} from '../../mix/types'
+import chalk from 'chalk'
+import {channelColors} from '../../mix/api/utils/channel-colors'
+
+const debug = makeDebug('mix:commands:channels:create')
+
+export default class ChannelsCreate extends MixCommand {
+  // TODO: longer description here
+  static description = 'create a new channel in a project'
+
+  static examples = [
+    'mix channels:create -P 1922 --name "New IVR channel" --mode ivr',
+  ]
+
+  static flags = {
+    project: MixFlags.projectFlag,
+    name: flags.string({
+      description: 'channel name',
+      required: true,
+    }),
+    mode: MixFlags.modesFlag,
+    color: flags.string({
+      description: 'channel color',
+    }),
+  }
+
+  get domainOptions(): DomainOption[] {
+    debug('get domainOptions()')
+    return ['project']
+  }
+
+  tryDomainOptionsValidation(options: any, domainOptions: DomainOption[]) {
+    debug('tryDomainOptionsValidation()')
+    super.tryDomainOptionsValidation(options, domainOptions)
+
+    // Check if all modes are valid and appear exactly once
```

Comment:
Align the validation with that of channels:configure. Extract common validation functionality to validations.ts or another helper file.